Fix bad redirection after error in address controller - edit usecase

matks opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | PR https://github.com/PrestaShop/PrestaShop/pull/20263 fixed the "create an address" usecase described in https://github.com/PrestaShop/PrestaShop/issues/20170 but not the "edit an address"
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/20170
| How to test?  | See ticket, for "edit an address" usecase. The PR https://github.com/PrestaShop/PrestaShop/pull/20263 fixed the "create an address" usecase. <br> Before the PR, you are redirected into "adresses listing" page. You should stay on the form with error message. This is what this PR fixes.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/20303)
<!-- Reviewable:end -->

matks committed:
Fix bad redirection after error in address controller - part 2

khouloudbelguith commented:
Hi @matks,

I tried with this PR fresh install and the problem still exists
I attached a screen record
https://drive.google.com/file/d/1gg8-vqlH8VWDZuR7aQLcSZZ-7mWTfNty/view?usp=sharing

Thanks!

matks committed:
Fix bad redirection after error in address controller - part 3

matks commented:
> Hi @matks,
> 
> I tried with this PR fresh install and the problem still exists
> I attached a screen record
> https://drive.google.com/file/d/1gg8-vqlH8VWDZuR7aQLcSZZ-7mWTfNty/view?usp=sharing
> 
> Thanks!

Hi @khouloudbelguith you are right !

After looking deeper at the code, I understood I fixed **another bug** instead of https://github.com/PrestaShop/PrestaShop/issues/20170

So
- I created an issue for the new bug => https://github.com/PrestaShop/PrestaShop/issues/20323
- I have updated this PR to solve https://github.com/PrestaShop/PrestaShop/issues/20170 AND #20323

khouloudbelguith commented:
Hi @matks,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/12G8w14HeyRc07AJfQHl-3tieeuymFYKT/view?usp=sharing
Thanks!

Progi1984 commented:
Thanks @matks & @khouloudbelguith
